### PR TITLE
KAFKA-16293: Test log directory failure in Kraft

### PR DESCRIPTION
**Pull request by gaurav-narula**
Enables log directory failure system test for all Kraft modes in addition to ZK mode.

This depends on #15335 being merged.

### CODE REVIEW

**Comment by gaurav-narula** [comment]
CC: @soarez @OmniaGM @pprovenzano @cmccabe

File: tests/kafkatest/tests/core/log_dir_failure_test.py
Comment (soarez) [inline tests/kafkatest/tests/core/log_dir_failure_test.py]: Unsure about the value of repeating the tests for `quorum.combined_kraft`, this mode isn't recommended for production, it's easier to cover it in java integration tests and makes this suite slower.

**Review by soarez — APPROVED** [review]
LGTM

File: tests/kafkatest/tests/core/log_dir_failure_test.py
Comment (pprovenzano) [reply tests/kafkatest/tests/core/log_dir_failure_test.py]: I agree with dropping the combined test here. I'm not sure I would ever suggest running JBOD in combined mode. A separate process for broker and controller on the same node would be better.

**Review by pprovenzano — APPROVED** [review]
LGTM with one comment mirroring Igor's comment.

**Comment by soarez** [comment]
@rondagostino @cmccabe can one of you have a look? Thanks

**Comment by soarez** [comment]
@showuon could you have a look at this?

**Review by showuon — APPROVED** [review]
LGTM! Verifying if the tests passed.

**Comment by showuon** [comment]
@gaurav-narula , the new added KRaft tests are failing in our test env. The logs are [here](https://drive.google.com/file/d/1CbxgH8eswEXX0YDEJpZn7B9Vd4x-gfo4/view?usp=sharing). Could you help check and fix them? Thanks.

**Comment by OmniaGM** [comment]
> @gaurav-narula , the new added KRaft tests are failing in our test env. The logs are [here](https://drive.google.com/file/d/1CbxgH8eswEXX0YDEJpZn7B9Vd4x-gfo4/view?usp=sharing). Could you help check and fix them? Thanks.

I believe this pr depends on https://github.com/apache/kafka/pull/15335

**Comment by showuon** [comment]
> > @gaurav-narula , the new added KRaft tests are failing in our test env. The logs are [here](https://drive.google.com/file/d/1CbxgH8eswEXX0YDEJpZn7B9Vd4x-gfo4/view?usp=sharing). Could you help check and fix them? Thanks.
> 
> I believe this pr depends on #15335

OK, let's wait until #15335 merged and then check again. Thanks.

**Comment by showuon** [comment]
Confirmed all tests passed after https://github.com/apache/kafka/pull/15335 merged.

**Pull request merged**
